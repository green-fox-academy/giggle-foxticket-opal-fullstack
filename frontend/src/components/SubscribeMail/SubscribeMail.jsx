import React, { useState } from 'react'
import mailImg from '../../assets/images/default-mail.svg'
import { FaExclamationTriangle } from 'react-icons/fa';
import './SubscribeMail.styles.sass'

export default function () {
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [validInput, setvalidInput] = useState(true);
  const [successMessage, setSuccessMessage] = useState({});
  
 
  const handleSubmit = e => {
    e.preventDefault();
    setSuccessMessage('')
    setvalidInput(true)
    if (name.length <= 3 || email.length <= 3) {
      setvalidInput(false);
      setSuccessMessage({})
    }else {
      const subscriber = { name, email };
      fetch('http://localhost:3000/api/subscription', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(subscriber)
      })
      .then(response => setSuccessMessage({statusText: response.statusText , statusCode: response.status}))
    }
  }
  
  const handleNameChange = (e) =>{
    setName(e.target.value)
  }

  const handleEmailChange = (e) =>{
    setEmail(e.target.value)
  }

  return (
    <form onSubmit={handleSubmit} className="subscription-form">
      <label>Subscribe to our newsletter!</label>
      <div className="emailImg">
        <img src={mailImg} alt="subscribe email"></img>
      </div>
      <div className="form-group">
        <input
          name="name" 
          placeholder="Your name"
          type="text"
          value={name}
          required
          onChange={handleNameChange}
        />
      </div>
      <div className="form-group">
        <input
          name="email"
          placeholder="Your email"
          type="email"
          value={email}
          required
          onChange={handleEmailChange}
        />
        { !validInput && 
        <p>
           Username and Email should be minimum 3 characters    
          <FaExclamationTriangle color="red" size="1.5em" />
          <i className="fas fa-exclamation-triangle"></i>
        </p>
        }
      </div>
      { successMessage.statusText &&  <p className={successMessage.statusCode === 201 ? 'msg success' : 'msg error'}>{successMessage.statusText}</p> }
      <button
        type="submit"
        className="btn">
        Subscribe!
      </button>  
    </form>      
  )
}
