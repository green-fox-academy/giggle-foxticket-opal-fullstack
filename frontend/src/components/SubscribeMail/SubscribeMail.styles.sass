.subscription-form
  display: flex
  flex-direction: column
  align-items: center
  margin-bottom: 5% 
  text-align: center

  label
    font-size: 25px
  
  .msg
    display: block
    margin-top: 20px
    font-size: 25px
    margin-bottom: 0
    &.success
      color: #3cb878d9
    &.error
      color: red
    &.success::after
      content: 'Köszönjük feliratkozását!'
    &.error::after
      content: 'Valahol hiba lépett fel a feliratkozás során! :('
    
  input
    margin-top: 10px
    min-width: 15em
    height: 30px
    padding: 0px 10px
    font-size: 16px
    background: #f1f1f1
    border: 0
    border-radius: 4px
    transition: all 250ms ease-in-out
    &:focus
      outline: none
      box-shadow: 0px 0px 12px 0.8px #3cb878d9

  .btn
    margin-top: 20px
    font-size: 20px
    padding: 7px 20px 9px 20px
    border: 0
    background: #3cb878d9
    color: white
    border-radius: 4px
    transition: all 350ms ease-in-out
    cursor: pointer
    &:hover 
      filter: brightness(115%)
    &:focus 
      outline: none

  .emailImg
    width: 150px
    height: 150px
    img
      height: 100%
      width: 100%

  .isValid
    display: none


