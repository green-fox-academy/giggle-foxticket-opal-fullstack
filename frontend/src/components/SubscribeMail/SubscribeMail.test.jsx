import React from 'react';
import SubscribeMail from './SubscribeMail';
import renderer from 'react-test-renderer';

describe('test SubscribeMail component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SubscribeMail />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
