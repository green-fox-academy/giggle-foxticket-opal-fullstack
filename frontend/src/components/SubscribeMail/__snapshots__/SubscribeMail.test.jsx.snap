// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SubscribeMail component renders correctly 1`] = `
<form
  className="subscription-form"
  onSubmit={[Function]}
>
  <label>
    Subscribe to our newsletter!
  </label>
  <div
    className="emailImg"
  >
    <img
      alt="subscribe email"
      src="default-mail.svg"
    />
  </div>
  <div
    className="form-group"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Your name"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Your email"
      required={true}
      type="email"
      value=""
    />
  </div>
  <button
    className="btn"
    type="submit"
  >
    Subscribe!
  </button>
</form>
`;
